zebra/zebra_rib.c

```diff
@@ -2704,15 +2704,9 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		apply_mask_ipv6(src_p);
 
 	/* Set default distance by route type. */
-	if (re->distance == 0) {
+	if (re->distance == 0)
```

Comment:
In the other commit, ZEBRA_EBGP_DISTANCE_DEFAULT is assigned to api.distance which is 20, but not 200. 

The check removed explicitly set the distance for IBGP case and to 200. So how does this work ? doesn't an IBGP route end up with distance 20 with this change ?

Reply:
no because every other route install from bgp into zebra correctly sets the admin distance.  This one did not.   look at bgp_zebra_route_announe

Reply:
Donald, Not clear to me. Will ping you on slack. Thanks.

Reply:
> In the other commit, ZEBRA_EBGP_DISTANCE_DEFAULT is assigned to api.distance which is 20, but not 200.
> 
> The check removed explicitly set the distance for IBGP case and to 200. So how does this work ? doesn't an IBGP route end up with distance 20 with this change ?

`bgp_zebra_announce_default()` doesn't set the `ZEBRA_FLAG_IBGP` flag in the message, so the announced route would always get the eBGP default distance in zebra. That commit only made that more explicit.